core/src/main/java/edu/wpi/grip/core/events/ConnectionAddedEvent.java

```diff
@@ -3,18 +3,20 @@
 import com.google.common.base.MoreObjects;
 import edu.wpi.grip.core.Connection;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+
 /**
  * An event that occurs when a new connection is added to the pipeline.  This is triggered by the user adding a
  * connection with the GUI.
  */
-public class ConnectionAddedEvent {
+public class ConnectionAddedEvent implements RunPipelineEvent {
     private final Connection connection;
 
     /**
      * @param connection The connection being added
      */
     public ConnectionAddedEvent(Connection connection) {
-        this.connection = connection;
+        this.connection = checkNotNull(connection, "Connection can not be null");
```

Comment:
Nitpicking, but *cannot. "can not" usually indicates that "not" is modifying the word after it.